addons/sale_pdf_quote_builder/models/ir_actions_report.py

```diff
@@ -24,50 +24,105 @@ def _render_qweb_pdf_prepare_streams(self, report_ref, data, res_ids=None):
-                        or line.product_template_id.product_document_ids
+                    product_product_docs = line.product_id.product_document_ids
+                    product_template_docs = line.product_template_id.product_document_ids
+                    doc_to_include = (
+                        product_product_docs.filtered(lambda d: d.attached_on == 'inside')
+                        or product_template_docs.filtered(lambda d: d.attached_on == 'inside')
                     )
-                    doc_to_include = documents.filtered(lambda d: d.attached_on == 'inside')
                     included_product_docs = included_product_docs | doc_to_include
 
-                if not header and not included_product_docs and not footer:
+                if (not has_header and not included_product_docs and not has_footer):
                     continue
 
                 IrBinary = self.env['ir.binary']
+                so_form_fields = self._get_so_form_fields_mapping(order)
                 pdf_data = []
-                if header:
-                    header_stream = IrBinary._record_to_stream(record, 'sale_header').read()
+                if has_header:
+                    header_stream = IrBinary._record_to_stream(header_record, 'sale_header').read()
+                    header_stream = pdf.fill_form_fields_pdf(header_stream, so_form_fields)
                     pdf_data.append(header_stream)
-                for included_doc in included_product_docs:
-                    doc_stream = IrBinary._record_to_stream(included_doc, 'datas').read()
-                    pdf_data.append(doc_stream)
+                if included_product_docs:
+                    docs_streams = self._fill_sol_documents_fields(
+                        order, included_product_docs, so_form_fields
+                    )
+                    pdf_data.extend(docs_streams)
+
                 pdf_data.append((initial_stream).getvalue())
-                if footer:
-                    footer_stream = IrBinary._record_to_stream(record, 'sale_footer').read()
+                if has_footer:
+                    footer_stream = IrBinary._record_to_stream(footer_record, 'sale_footer').read()
+                    footer_stream = pdf.fill_form_fields_pdf(footer_stream, so_form_fields)
                     pdf_data.append(footer_stream)
 
-                try:
-                    form_fields = self._get_form_fields_mapping(order)
-                    stream = io.BytesIO(pdf.merge_pdf(pdf_data, form_fields))
-                    result[order.id].update({'stream': stream})
-                except (ValueError, PdfStreamError, PdfReadError, TypeError, zlib_error,
-                        NotImplementedError):
-                    order._message_log(body=_(
-                        "There was an error when trying to merge headers and footers to the "
-                        "original PDF.\n Please make sure the source file are valid."
-                    ))
```

Comment:
No more feedback on errors ?

Reply:
This was added when the feature was bugged + when it was still possible to put a non pdf file, which crashed at this point. But I only had to cover one method with this try except. Now the filling in the forms with so information, or sol information, or merging it is separated, adding this error every time is quite heavy, and I'm not sure of the added value of this.  There could still be some errors, obviously, but I'm not sure which ones I should track to cover everything related, and I'm a bit afraid to cover too much and silence something that I shouldn't.